hbase-client/src/main/java/org/apache/omid/transaction/SnapshotFilterImpl.java

```diff
@@ -298,26 +291,62 @@ public CommitTimestamp locateCellCommitTimestamp(long cellStartTimestamp, long e
                     CellUtil.matchingValue(cell, HConstants.EMPTY_BYTE_ARRAY)) {
-
                 String row = Bytes.toString(cell.getRow());
-                List<Cell> cells = familyDeletionCache.get(row);
-                if (cells == null) {
-                    cells = new ArrayList<>();
-                    familyDeletionCache.put(row, cells);
+                String family = Bytes.toString(cell.getFamily());
+                String key = row + ":" + family;
+
+                if (familyDeletionCache.containsKey(key))
+                    return;
+
+                Optional<Long> commitTimeStamp = isCellInSnapshot(cell, transaction, commitCache);
+
+                if (!commitTimeStamp.isPresent()) {
+                    commitTimeStamp = isCellInTransaction(cell, transaction, commitCache);
                 }
 
-                cells.add(cell);
+                if (commitTimeStamp.isPresent()) {
+                    familyDeletionCache.put(key, commitTimeStamp.get());
+                } else {
+                    Cell lastCell = cell;
+                    Map<Long, Long> cmtCache;
+                    boolean foundCommitttedFamilyDeletion = false;
+                    while (!foundCommitttedFamilyDeletion) {
+
+                        Get g = createPendingGet(lastCell, 3);
+                        for (Map.Entry<String,byte[]> entry : attributeMap.entrySet()) {
+                            g.setAttribute(entry.getKey(), entry.getValue());
+                        }
+
+                        Result result = tableAccessWrapper.get(g);
+                        List<Cell> resultCells = result.listCells();
+                        if (resultCells == null) {
+                            break;
+                        }
+
+                        cmtCache = buildCommitCache(resultCells);
+                        for (Cell c : resultCells) {
+                            if (CellUtil.matchingQualifier(c, CellUtils.FAMILY_DELETE_QUALIFIER) &&
+                                    CellUtil.matchingValue(c, HConstants.EMPTY_BYTE_ARRAY)) {
+                                    commitTimeStamp = isCellInSnapshot(c, transaction, cmtCache);
+                                    if (commitTimeStamp.isPresent()) {
+                                        familyDeletionCache.put(key, commitTimeStamp.get());
+                                        foundCommitttedFamilyDeletion = true;
+                                        break;
+                                    }
+                                    lastCell = c;
+                            }
+                        }
+                    }
+                }
             }
         }
-
     }
 
-    private boolean isCellInTransaction(Cell kv, HBaseTransaction transaction, Map<Long, Long> commitCache) {
+    private Optional<Long> isCellInTransaction(Cell kv, HBaseTransaction transaction, Map<Long, Long> commitCache) {
```

Comment:
rename function because we dont return boolean now.
maybe:
"getTSIfInTransaction"